public/views/content/student.html

```diff
@@ -9,6 +9,14 @@
     </p>
   </div>
 </div>
+<div ng-switch="$state.current.name.includes('showd3')">
+  <div ng-switch-when="true">
+    <a ui-sref=".">- Show Visualizations</a>
+  </div>
+  <div ng-switch-when="false">
+    <a ui-sref=".showd3" ng-click = "showData(currentStudent)">+ Show Visualizations</a>
```

Comment:
I think the point of using `ui-sref` is that we don't have to use `ng-click`. Basically, everything that needs to happen after the click can just be in the controller definition.